aspects/request_timeout_handler_test.go

```diff
@@ -0,0 +1,57 @@
+package aspects
+
+import (
+	"github.com/julienschmidt/httprouter"
+	"github.com/prebid/prebid-server/config"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var reqTimeInQueueHeaderName = "X-Ngx-Request-Time"
+var reqTimeoutHeaderName = "X-Ngx-Request-Timeout"
+
+func TestQueuedRequestTimeoutWithTimeout(t *testing.T) {
+
+	rw := ExecuteAspectRequest(t, "6")
+
+	assert.Equal(t, http.StatusRequestTimeout, rw.Code, "Http response code is incorrect, should be 408")
+
+}
+
+func TestQueuedRequestTimeoutNoTimeout(t *testing.T) {
+
+	rw := ExecuteAspectRequest(t, "0.9")
+
+	assert.Equal(t, http.StatusOK, rw.Code, "Http response code is incorrect, should be 200")
+
+}
+
+func MockEndpoint() httprouter.Handle {
+	return httprouter.Handle(MockHandler)
+}
+
+func MockHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {}
+
+func ExecuteAspectRequest(t *testing.T, timeout string) *httptest.ResponseRecorder {
```

Comment:
This is minor since it is a test function, but it seems `timeout` argument represents the time in queue. Consider renaming to `timeInQueue` or something similar.

Reply:
Fixed